core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/identity/request/GenerateApiKeyRequest.java

```diff
@@ -65,28 +66,42 @@ void setExpressions(List<String> permissions) {
 	
 	@Override
 	public User execute(ServiceProvider context) {
-		User user = null; 
+		User user = null;
+		List<Permission> newTokenPermissions = List.of();
 		
 		if (!Strings.isNullOrEmpty(username) && !Strings.isNullOrEmpty(password)) {
 			user = context.service(IdentityProvider.class).auth(username, password);
+			newTokenPermissions = this.permissions == null ? user.getPermissions() : this.permissions.stream().map(Permission::valueOf).collect(Collectors.toList());
 		} else if (!Strings.isNullOrEmpty(token)) {
 			user = context.service(AuthorizationHeaderVerifier.class).authJWT(token);
+			// in case of token based auth, we only allow refreshing the token with the same permissions, to generate new tokens, the user/client has to provide its username and password again
+			if (!CompareUtils.isEmpty(this.permissions)) {
+				throw new BadRequestException("Token cannot be refreshed with a different permission claim value.");
```

Comment:
The message suggests that a comparison took place and token generation would've completed successfully, had `permissions` contained the exact same elements (or a subset) as the submitted token.

Reply:
Done.